lib/improver/wind_direction.py

```diff
@@ -0,0 +1,409 @@
+                                r_vals, r_thresh, realization_axis):
+        """Find standard deviation of polar numbers.
+
+        The average wind direction complex values represent the midpoint
+        between the different values and so have r values between 0-1.
+
+        1) From wind_dir_deg_mean - create a new set of complex values.
+           Therefore they will have the same angle but r is fixed as r=1.
+        2) Use fixed-r average complex values to find the distance between the
+           mean point and all the wind direction values
+        3) Find the average distance between the mean point and the wind
+           direction values. Large average distance == low confidence.
+        4) Normalise the average distance values to produce a confidence
+           value that is equivalent to the standard deviation
+
+        Args:
+            wind_dir_complex (np.ndarray):
+                3D array - wind direction angles in complex numbers.
+            wind_dir_deg_mean (np.ndarray):
+                3D array - average wind direction in angles.
+            r_vals (np.ndarray):
+                3D array - Radius taken from average complex wind direction
+                angle.
+            r_thresh (float):
+                Any r value below threshold is regarded as meaningless.
+            realization_axis (int):
+                Axis over which to average the arrays over.
+
+        Returns:
+            dist_from_mean_norm (np.ndarray):
+                3D array - The average distance from mean normalised - used
+                as a confidence value.
+        """
+
+        wind_dir_complex_mean_fixr = WindDirection.deg_to_complex(
+            wind_dir_deg_mean)
+
+        # Find difference in the distance between all the observed points and
+        # mean point with fixed r=1.
+        # For maths to work - the "wind_dir_complex_mean_fixr array" needs to
+        # be "tiled" so that it is the same dimension as "wind_dir_complex".
+        wind_dir_complex_mean_tile = np.tile(wind_dir_complex_mean_fixr,
+                                             (wind_dir_complex.shape[0], 1, 1))
+
+        # Calculate distance from each wind direction data point to the
+        # average point.
+        difference = wind_dir_complex - wind_dir_complex_mean_tile
+        dist_from_mean = np.sqrt(np.square(difference.real) +
+                                 np.square(difference.imag))
+
+        # Find average distance.
+        dist_from_mean_avg = np.mean(dist_from_mean, axis=realization_axis)
+
+        # If we have two points at opposite ends of the compass
+        # (eg. 270 and 90), then their separation distance is 2.
+        # Normalise the array using 2 as the maximum possible value.
+        dist_from_mean_norm = 1 - dist_from_mean_avg*0.5
+
+        # With two points directly opposte (270 and 90) it returns a confidence
+        # value of 0.29289322 instead of zero.
```

Comment:
This is a precision error because r is very small.  We need to set the confidence values when r is sufficiently small (smaller than chosen threshold) to 0